File: lib/server/features/operation_logs/operation_log.dart
class OperationLog {
  final int? id;
  final String category;
  final String action;
  final String? target;
  final Map<String, dynamic>? detail;
  final String status;
  final String? errorMessage;
  final String? connectionId;
  final int createdAt;

  OperationLog({
    this.id,
    required this.category,
    required this.action,
    this.target,
    this.detail,
    required this.status,
    this.errorMessage,
    this.connectionId,
    required this.createdAt,
  });

  Map<String, dynamic> toJson() => {
    'id': id,
    'category': category,
    'action': action,
    'target': target,
    'detail': detail,
    'status': status,
    'errorMessage': errorMessage,
    'connectionId': connectionId,
    'createdAt': createdAt,
  };
}

File: host-deck-ui/src/api/operation-log.ts
import { http } from '@/lib/http'

export type OperationLogStatus = 'success' | 'failed'

export interface OperationLogItem {
  id: number
  category: string
  action: string
  target?: string | null
  detail?: Record<string, unknown> | null
  status: OperationLogStatus
  errorMessage?: string | null
  connectionId?: string | null
  createdAt: number
}

export interface OperationLogListParams {
  category?: string
  limit?: number
  offset?: number
  status?: OperationLogStatus | ''
}

export const operationLogApi = {
  async clear() {
    const response = await http.delete<{ success: boolean }>('/api/operation-logs')
    return response.data
  },

  async list(params: OperationLogListParams = {}) {
    const response = await http.get<OperationLogItem[]>('/api/operation-logs', { params })
    return response.data
  },
}
